hyperion/app/creation_steps/workspace/git_manager.py

```
@@ -0,0 +1,725 @@
+                if result.returncode != 0:
+                    logger.warning(
+                        f"Failed to add pattern '{pattern}': {result.stderr}"
+                    )
+                else:
+                    logger.debug(f"Added pattern '{pattern}' to staging area")
+
+            logger.info("Files added to Git staging area successfully")
+
+        except Exception as e:
+            logger.error(f"Failed to add files to Git: {e}")
+            raise GitException(f"Failed to add files to Git: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)